Writerside/topics/DualCodeModelAPIs.md

```diff
@@ -1,3 +1,15 @@
 # The Dual Code Model APIs
 
-_To be written_
\ No newline at end of file
+This segment is more theoretical than the others. It is meant to provide a high-level overview of the dual code model APIs and the approach followed in the development of the starlasu tools.
+It will approach the concepts of homogenous and heterogeneous APIs, and how to use and leveraged them in the starlasu tools.
+
+## Homogenous APIs
+
+In kolasu, [Nodes](https://github.com/Strumenta/kolasu/blob/main/core/src/main/kotlin/com/strumenta/kolasu/model/Node.kt) are the basic building blocks of an AST. They are used to represent the different elements of the language being parsed.
+All the instances of nodes are the same and have a defined set of properties and methods, such as the origin, parent, etc... This is what we call a homogenous API. 
```

Comment:
I would mention the reflective capabilities: we can ask each node for each properties and for each property we know the name, the type, the multiplicity. This allows to write fully generic algorithms.